internal/ratelimit/handler.go

```diff
@@ -127,21 +148,23 @@ func Handler(l *rate.Limiter, next http.Handler) http.Handler {
 			return
 		}
 
-		allowed, quota, err := l.Allow(res, a)
+		allowed, quota, err := l.Allow(res, a, reqInfo.ClientIp, authtoken)
 		if err != nil {
 			if errFull, ok := err.(*rate.ErrLimiterFull); ok {
 				rw.Header().Add("Retry-After", fmt.Sprintf("%.0f", math.Ceil(errFull.RetryIn.Seconds())))
 				rw.WriteHeader(http.StatusServiceUnavailable)
 				return
 			}
-
 			// The only other error here should be rate.ErrLimitNotFound, which
 			// shouldn't be possible given how we initialize the limiter and
 			// the checks done by extractResourceAction.
 			rw.WriteHeader(http.StatusInternalServerError)
 			return
 		}
 
+		l.SetUsageHeader(quota, rw.Header())
+		l.SetPolicyHeader(res, a, rw.Header())
```

Comment:
`SetPolicyHeader` returns an error. Should we check it?

Reply:
The only thing I think we could do with an error from this would be wrap it so there is an error event. At least currently, we should never encounter an error here, since the only possible error from this we would have encountered earlier and returned from the handler before getting here.